net/Socket.cpp

```diff
@@ -1239,50 +1239,58 @@ std::shared_ptr<Socket> ServerSocket::accept()
 #else
     const int rc = fakeSocketAccept4(getFD());
 #endif
-    LOG_TRC("Accepted socket #" << rc << ", creating socket object.");
-    try
+    if (rc < 0)
     {
-        // Create a socket object using the factory.
-        if (rc != -1)
+        const int cause = errno;
+        if( cause != EINTR && cause != EAGAIN && cause != EWOULDBLOCK )
```

Comment:
It seems to me that leaving coolwsd running without an externally listening socket is a bad move here. 
Can we restore the nesting of the previous 'try' - to avoid all this code change / movement.

And instead do a LOG_FTL() and a Util::forcedExit(EX_SOFTWARE); in the case that we get an unrecoverable error in this case.

Then I expect we can minimize the change, the diff, reduce review time & do something more sensible than before.